examples/playwright/src/theia-notebook-editor.ts

```diff
@@ -0,0 +1,164 @@
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Locator } from '@playwright/test';
+import { join } from 'path';
+import { TheiaApp } from './theia-app';
+import { TheiaEditor } from './theia-editor';
+import { TheiaNotebookCell } from './theia-notebook-cell';
+import { TheiaNotebookToolbar } from './theia-notebook-toolbar';
+import { TheiaQuickCommandPalette } from './theia-quick-command-palette';
+import { TheiaToolbarItem } from './theia-toolbar-item';
+import { OSUtil, normalizeId, urlEncodePath } from './util';
+
+export class TheiaNotebookEditor extends TheiaEditor {
+
+    constructor(filePath: string, app: TheiaApp) {
+        // shell-tab-notebook::file://<path>
+        // notebook:file://<path>
+        super({
+            tabSelector: normalizeId(`#shell-tab-notebook:file://${urlEncodePath(join(app.workspace.escapedPath, OSUtil.fileSeparator, filePath))}`),
+            viewSelector: normalizeId(`#notebook:file://${urlEncodePath(join(app.workspace.escapedPath, OSUtil.fileSeparator, filePath))}`)
+        }, app);
+    }
+
+    protected viewLocator(): Locator {
+        return this.page.locator(this.data.viewSelector);
+    }
+
+    tabLocator(): Locator {
+        return this.page.locator(this.data.viewSelector);
+    }
+
+    override async waitForVisible(): Promise<void> {
+        await super.waitForVisible();
+        // wait for toolbar being rendered as it takes some time to load the kernel data.
+        await this.notebookToolbar().waitForVisible();
+    }
+
+    /**
+     * @returns The main toolbar of the notebook editor.
+     */
+    notebookToolbar(): TheiaNotebookToolbar {
+        return new TheiaNotebookToolbar(this.viewLocator(), this.app);
+    }
+
+    /**
+     * @returns The name of the selected kernel.
+     */
+    async selectedKernel(): Promise<string | undefined | null> {
+        const kernelItem = await this.toolbarItem('notebook.selectKernel');
+        if (!kernelItem) {
+            throw new Error('Select kernel toolbar item not found.');
+        }
+        return this.notebookToolbar().locator.locator('#kernel-text').innerText();
+    }
+
+    /**
+     *  Allows to select a kernel using toolbar item.
+     * @param kernelName  The name of the kernel to select.
+     */
+    async selectKernel(kernelName: string): Promise<void> {
+        // TODO should we use NotebookCommands.SELECT_KERNEL_COMMAND.id? Need a dependency...
```

Comment:
Good question, I think I tend to avoid this dependency but rather introduce a namespace with constants here in the Playwright page object library.